subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/StaleOutputIntegrationTest.groovy

```diff
@@ -436,18 +434,29 @@ class StaleOutputIntegrationTest extends AbstractIntegrationSpec {
         run 'backup', 'restore'
 
         then:
+        executedAndNotSkipped(':backup')
+        executedAndNotSkipped(':restore')
         original.text == backup.text
         original.text == "Original"
 
         when:
-        changeGradleVersion()
-        run 'backup', 'restore'
+        /*
+        The whole setup is as follows:
+        - Both the restore and the backup task have an entry in the task history
+        - The output of the restore path is the input of the backup task.
+          This means when the backup task is up-to-date, then the contents of the output of the restore path is in the file system mirror.
+
+        If cleaning up stale output files does not invalidate the file system mirror, then the restore task would be up-to-date.
+         */
```

Comment:
Style: I think we shouldn't be using `/* ... */`-style comments.

Reply:
I'm still wondering if we could come up with an easier-to-understand test here if the two tasks were not locked in a circle. Like, we could trigger stale outputs just by creating a file among the outputs of the task between executions.

Reply:
We could create the file, but the file would not be in the file system mirror if no other task snapshots it. This is why I need the `backup` task to have the output of the `restore` task as in input.

Reply:
I replaced `/*` by `//`.